lib/cli-engine.js

```diff
@@ -536,12 +536,18 @@ CLIEngine.prototype = {
 
         var results = [],
             stats,
+            globOptions,
             options = this.options,
             configHelper = new Config(options),
-            ignoredPaths = IgnoredPaths.load(options),
-            exclude = ignoredPaths.contains.bind(ignoredPaths);
+            ignoredPaths = options.ignore ? IgnoredPaths.load(options).patterns : [];
+
+        globOptions = {
+            nodir: true,
+            realpath: true,
+            ignore: ignoredPaths.map(processPath)
+        };
 
-        if (filename && options.ignore && exclude(filename)) {
+        if (filename && options.ignore && glob.sync(filename, globOptions).length === 0) {
```

Comment:
Isn’t this more inefficient than the previous `minimatch` implementation? Looks like `glob.sync` is a blocking I/O function and `minimatch` only compares strings via regexp.